crates/bevy_pbr/src/render/mesh.rs

```diff
@@ -317,10 +337,124 @@ pub fn extract_meshes(
+                "_depth"
+            } else {
+                ""
+            },
+            if self.contains(MeshPipelineViewLayoutKey::NORMAL_PREPASS) {
+                "_normal"
+            } else {
+                ""
+            },
+            if self.contains(MeshPipelineViewLayoutKey::MOTION_VECTOR_PREPASS) {
+                "_motion"
+            } else {
+                ""
+            },
+            if self.contains(MeshPipelineViewLayoutKey::DEFERRED_PREPASS) {
+                "_deferred"
+            } else {
+                ""
+            },
+        )
+    }
+}
+
+impl From<MeshPipelineKey> for MeshPipelineViewLayoutKey {
+    fn from(value: MeshPipelineKey) -> Self {
+        let mut result = MeshPipelineViewLayoutKey::empty();
+
+        if value.msaa_samples() > 1 {
+            result |= MeshPipelineViewLayoutKey::MULTISAMPLED;
+        }
+        if value.contains(MeshPipelineKey::DEPTH_PREPASS) {
+            result |= MeshPipelineViewLayoutKey::DEPTH_PREPASS;
+        }
+        if value.contains(MeshPipelineKey::NORMAL_PREPASS) {
+            result |= MeshPipelineViewLayoutKey::NORMAL_PREPASS;
+        }
+        if value.contains(MeshPipelineKey::MOTION_VECTOR_PREPASS) {
+            result |= MeshPipelineViewLayoutKey::MOTION_VECTOR_PREPASS;
+        }
+        if value.contains(MeshPipelineKey::DEFERRED_PREPASS) {
+            result |= MeshPipelineViewLayoutKey::DEFERRED_PREPASS;
+        }
+
+        result
+    }
+}
+
+impl From<Msaa> for MeshPipelineViewLayoutKey {
+    fn from(value: Msaa) -> Self {
+        let mut result = MeshPipelineViewLayoutKey::empty();
+
+        if value.samples() > 1 {
+            result |= MeshPipelineViewLayoutKey::MULTISAMPLED;
+        }
+
+        result
+    }
+}
+
+impl From<Option<&ViewPrepassTextures>> for MeshPipelineViewLayoutKey {
+    fn from(value: Option<&ViewPrepassTextures>) -> Self {
+        let mut result = MeshPipelineViewLayoutKey::empty();
+
+        if let Some(prepass_textures) = value {
+            if prepass_textures.depth.is_some() {
+                result |= MeshPipelineViewLayoutKey::DEPTH_PREPASS;
+            }
+            if prepass_textures.normal.is_some() {
+                result |= MeshPipelineViewLayoutKey::NORMAL_PREPASS;
+            }
+            if prepass_textures.motion_vectors.is_some() {
+                result |= MeshPipelineViewLayoutKey::MOTION_VECTOR_PREPASS;
+            }
+            if prepass_textures.deferred.is_some() {
+                result |= MeshPipelineViewLayoutKey::DEFERRED_PREPASS;
+            }
+        }
+
+        result
+    }
+}
+
+#[derive(Clone)]
+struct MeshPipelineViewLayout {
+    pub bind_group_layout: BindGroupLayout,
+
+    #[cfg(debug_assertions)]
+    pub texture_count: usize,
+}
+
 #[derive(Resource, Clone)]
 pub struct MeshPipeline {
-    pub view_layout: BindGroupLayout,
-    pub view_layout_multisampled: BindGroupLayout,
+    view_layouts: [MeshPipelineViewLayout; 32],
```

Comment:
Does the following compile?
```suggestion
    view_layouts: [MeshPipelineViewLayout; MeshPipelineViewLayoutKey::COUNT],
```

Reply:
Nice catch, I had updated it in the other location but forgot to change it here.